docs/sphinx/heps/hep1.rst

```diff
@@ -0,0 +1,54 @@
+HEP 1: |hpx| build configurations and exported libraries
+========================================================
+
+The motivation is to provide smaller parts of |hpx| for users to consume. This
+is meant to address both use cases where a user builds |hpx| manually and and
+wishes to build a subset of |hpx|, and cases where a user uses a preinstalled
+|hpx| (distro packages, cluster modules) which may have most features enabled
+but still only wishes to pull in a subset of |hpx|.
+
+|hpx| will provide the following targets (and libraries):
+
+- ``HPX::core``: thread pools, schedulers, coroutines, and utilities
+- ``HPX::local``: depends on ``HPX::core`` (``hpx::thread``, async, algorithms,
+  local LCOs)
+- ``HPX::hpx``: depends on everything above, includes local and distributed
+  runtime, and distributed LCOs
+
+Separate shared libraries ("components"). These are all extensions on top of
+|hpx|:
+
+- ``HPX::cuda``
+- ``HPX::rocm`` (maybe soon?)
+- ``HPX::mpi``
+- parcelports as actual plugins again? (they are now statically linked to ``libhpx``)
+- ``HPX::iostreams_component`` and friends (drop the ``_component`` suffix?)
+
+Including headers from ``HPX::local`` does not pull in headers from
+``HPX::hpx``. The granularity at which features can be disabled at compile time
+in HPX would be these targets because it's a manageably small set of targets,
+unlike the modules we have now. That means the smallest part of HPX one can
+build is ``HPX::core``. If a user needs an even smaller part of HPX they can
+link to the static module libraries. They will have to build at least
+``HPX::core`` though.
+
+``HPX::hpx`` includes:
+
+- local runtime
+- distributed runtime
+- distributed LCOs
```

Comment:
For distributed HPX users: I'd be happy to split this further if there's a need for it, so please comment if you think that's the case. I'd start with this simply because this is simpler and I would guess few distributed HPX users would need only the LCOs but not the runtime (not even possible at the moment, but could be done in a similar way to how we get the default thread pool). I'm not a distributed HPX user so I rely on your feedback here.

Reply:
I don't think it makes much sense to split this even further for now.